lib/quickpay/api/client.rb

```diff
@@ -42,29 +50,48 @@ def initialize(username: nil, password: nil, base_uri: "https://api.quickpay.net
           end
 
           req = Request.new(
-            method,
+            method.to_sym,
             path,
             scrub_body(body.dup, headers["Content-Type"]),
             headers,
-            options.fetch(:query, {})
+            options[:query]
           ).freeze
 
-          res = @connection.request(**req.to_h)
-          error = QuickPay::API::Error.by_status_code(res.status, res.body, res.headers, req)
+          uri = URI(@base_uri)
+          uri.path << req.path
+          if (query = req.query) && query.any?
+            uri.query = URI.encode_www_form(req.query)
+          end
+          net_req = method_class.new(uri, req.headers)
+          net_req.basic_auth(@username, @password) if @username || @password
+          net_req.body = req.body
+          res = Net::HTTP.start(
+            uri.hostname,
+            use_ssl: uri.scheme == HTTPS,
+            open_timeout: @connect_timeout,
+            read_timeout: @read_timeout,
+            write_timeout: @write_timeout
+          ) do |http|
+            http.request(net_req)
+          end
+          status_code = res.code.to_i
+          body = res.body
+          headers = res.each_header.to_h
+          error = QuickPay::API::Error.by_status_code(status_code, body, headers, req)
 
-          if !options.fetch(:raw, false) && res.headers["Content-Type"] =~ CONTENT_TYPE_JSON_REGEX
-            res.body = JSON.parse(res.body, options[:json_opts] || @connection.data[:json_opts])
+          if !options.fetch(:raw, false) && res["content-type"] =~ CONTENT_TYPE_JSON_REGEX
```

Comment:
Can we be 100% sure the headers are always lowercase?

Reply:
Seems so https://github.com/ruby/net-http/blob/master/lib/net/http/header.rb#L40-L47 or at least they're treated as case-insensitive